### PR TITLE
[PLGN-220] Rest 

### PR DESCRIPTION
**Pull request by abmccann-r7**
### Description

Describe the proposed changes:
Changed error handling to prevent list has no attribute get error
  -

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/rest/Dockerfile [hunk 1]
```diff
@@ -1,29 +1,20 @@
...
-# Install plugin
-RUN python setup.py build && python setup.py install
+ADD . /python/src
+
+RUN	python setup.py build && python setup.py install
 
-# needed for SSL Verify
-USER root
+# User to run plugin code. The two supported users are: root, nobody
+USER nobody
```
Comment (TKuzas-R7) [inline plugins/rest/Dockerfile:18]: just double checking that changing to nobody wont break anything.
It looks like there was a comment here previously saying it was necessary for SSL Verify.

File: plugins/rest/Dockerfile [hunk 2]
```diff
@@ -1,29 +1,20 @@
...
 
-# Entrypoint
-ENTRYPOINT ["/usr/local/bin/komand_rest"]
+ENTRYPOINT ["/usr/local/bin/icon_rest"]
```
Comment (TKuzas-R7) [inline plugins/rest/Dockerfile:20]: Trailing New Line
```suggestion
ENTRYPOINT ["/usr/local/bin/icon_rest"]

```

File: plugins/rest/komand_rest/actions/delete/__init__.py [hunk 3]
```diff
@@ -1,2 +1,2 @@
-# GENERATED BY KOMAND SDK - DO NOT EDIT
-from .action import Delete
+# GENERATED BY INSIGHT-PLUGIN - DO NOT EDIT
+from .action import Delete
```
Comment (TKuzas-R7) [inline plugins/rest/komand_rest/actions/delete/__init__.py:2]:
```suggestion
from .action import Delete

```

Comment (TKuzas-R7) [reply plugins/rest/komand_rest/actions/delete/__init__.py:2]: The updated Insight tooling should stop this from happening in the future, be sure to update your tooling to the newest version.

**Review by TKuzas-R7 — CHANGES_REQUESTED** [review]

File: plugins/rest/Dockerfile (hunk 2 above)
Comment (TKuzas-R7) [reply plugins/rest/Dockerfile:20]: I'm not sure if the trailing new line actually affects much, but we tend to leave them in where we can, could you add them back where the tooling removed it?

File: plugins/rest/Dockerfile (hunk 1 above)
Comment (igorski-r7) [reply plugins/rest/Dockerfile:18]: Try to add `sdk` field to `plugin.spec.yaml` file. That would affect the `Dockerfile` 😄 . For example:
```
sdk:
  type: fyll
  version: 5
  user: root
  packages:
    - curl
    - another_package
    - etc...
```

File: plugins/rest/Dockerfile (hunk 2 above)
Comment (igorski-r7) [reply plugins/rest/Dockerfile:20]: It shouldn't be updated. The correct `ENTRYPOINT` here is `komand_rest` 😄

File: plugins/rest/help.md [hunk 4]
```diff
@@ -12,459 +11,320 @@ This plugin is often used to integrate with ad-hoc 3rd party API's in a workflow
 * A RESTFUL HTTP/HTTPS resource and supported authentication, if any
 
 # Supported Product Versions
-
-* 2022-09-05
+  
+* 2022-09-05
```
Comment (igorski-r7) [inline plugins/rest/help.md:15]: Those spaces needs to be removed

File: plugins/rest/help.md [hunk 5]
```diff
@@ -12,459 +11,320 @@ This plugin is often used to integrate with ad-hoc 3rd party API's in a workflow
...
 ##### Output
 
-|Name|Type|Required|Description|Example|
-| :--- | :--- | :--- | :--- | :--- |
-|body_object|object|False|Response payload from the server as an object. Note, if the response has invalid object structure (list, string..) plugin will wrap it with object map|{'user': 'user@example.com'}|
-|body_string|string|False|Response payload from the server as a string|test data|
-|headers|object|False|Response headers from the server|{'Host': 'rapid7.com'}|
-|status|integer|False|Status code of the response from the server|200|
-
+|Name|Type|Default|Required|Description|Enum|Example|
```
Comment (igorski-r7) [inline plugins/rest/help.md:91]: This needs to be updated in the tooling soon. No need to have that fields added.
```suggestion
|Name|Type|Required|Description|Example|
```

File: plugins/rest/help.md [hunk 6]
```diff
@@ -12,459 +11,320 @@ This plugin is often used to integrate with ad-hoc 3rd party API's in a workflow
...
+| :--- | :--- | :--- | :--- | :--- | :--- | :--- |
```
Comment (igorski-r7) [inline plugins/rest/help.md:92]:
```suggestion
| :--- | :--- | :--- | :--- | :--- |
```

File: plugins/rest/help.md [hunk 7]
```diff
@@ -12,459 +11,320 @@ This plugin is often used to integrate with ad-hoc 3rd party API's in a workflow
...
+|body_object|object|None|False|Response payload from the server as an object. Note, if the response has invalid object structure (list, string..) plugin will wrap it with object map|None|{'user': 'user@example.com'}|
+|body_string|string|None|False|Response payload from the server as a string|None|test data|
+|headers|object|None|False|Response headers from the server|None|{'Host': 'rapid7.com'}|
+|status|integer|None|False|Status code of the response from the server|None|200|
+
```
Comment (igorski-r7) [inline plugins/rest/help.md:199]: Same here

File: plugins/rest/help.md [hunk 8]
```diff
@@ -12,459 +11,320 @@ This plugin is often used to integrate with ad-hoc 3rd party API's in a workflow
...
```
Comment (igorski-r7) [inline plugins/rest/help.md:301]: Here as well 😄

File: plugins/rest/unit_test/payloads/put_array.json.resp
Comment (igorski-r7) [inline plugins/rest/unit_test/payloads/put_array.json.resp:1]: This one can be reformatted

**Review by igorski-r7 — CHANGES_REQUESTED** [review]

File: plugins/rest/plugin.spec.yaml [hunk 9]
```diff
@@ -13,6 +13,9 @@ resources:
   source_url: https://github.com/rapid7/insightconnect-plugins/tree/master/plugins/rest
   license_url: https://github.com/rapid7/insightconnect-plugins/blob/master/LICENSE
   docs_url: https://docs.rapid7.com/insightconnect/http-requests
+key_features: ["it can do this", "and that"]
```
Comment (cmcnally-r7) [inline plugins/rest/plugin.spec.yaml:16]: feel free to remove this (or change it) then regenerate 😄 
Now would be a good time to make use of this field properly. 
```suggestion
key_features: ["Make RESTful API calls"]
```

File: plugins/rest/plugin.spec.yaml [hunk 10]
```diff
@@ -13,6 +13,9 @@ resources:
...
+requirements: ["python3", "requests"]
```
Comment (cmcnally-r7) [inline plugins/rest/plugin.spec.yaml:17]: Requirements here does not refer to `requirements.txt` but rather what a user needs in order to use the plugin. In this case, they don't need anything technically, other than the details of the API they want to connect to.
So something like this:
```suggestion
requirements: ["None"]
```

File: plugins/rest/plugin.spec.yaml [hunk 11]
```diff
@@ -13,6 +13,9 @@ resources:
...
+references: ["www.google.com"]
```
Comment (cmcnally-r7) [inline plugins/rest/plugin.spec.yaml:18]:
```suggestion
references: ["https://requests.readthedocs.io/en/latest/", "https://docs.rapid7.com/insightconnect/http-requests"]
```

File: plugins/rest/help.md [hunk 12]
```diff
@@ -419,60 +301,39 @@ Example input:
...
+  
+*This plugin does not contain any custom output types.*  
 
 ## Troubleshooting
-
-Any headers set in the action will overwrite the default ones in the connection.
-
-Any issues connecting to the remote service should be present in the log of the job that ran. If you find any issues that represent bugs in the plugin itself, please contact someone at Komand directly.
+  
+*There is no troubleshooting for this plugin.*
```
Comment (cmcnally-r7) [inline plugins/rest/help.md:332]: lets change this back to how it was for now 
```suggestion
Any headers set in the action will overwrite the default ones in the connection.

Any issues connecting to the remote service should be present in the log of the job that ran. If you find any issues that represent bugs in the plugin itself, please contact someone at Komand directly.
```

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]
Looks good so far! Just a couple of suggested changes. Mostly issues with the tooling and the `refresh` command affecting the `help.md` rather than code functionality.

**Review by cmcnally-r7 — APPROVED** [review]
lgtm 🚀

**Review by igorski-r7 — APPROVED** [review]

File: plugins/rest/help.md [hunk 13]
```diff
@@ -500,11 +367,13 @@ Any issues connecting to the remote service should be present in the log of the
...
 
 * [HTTP Request Architecture Style](http://www.ics.uci.edu/~fielding/pubs/dissertation/rest_arch_style.htm)
-## References
 
-* [InsightConnect HTTP Request Plugin Guide](https://docs.rapid7.com/insightconnect/http-requests)
\ No newline at end of file
+## References
+  
+* https://requests.readthedocs.io/en/latest/  
+* https://docs.rapid7.com/insightconnect/http-requests
```
Comment (TKuzas-R7) [inline plugins/rest/help.md:379]:
```suggestion
* [InsightConnect HTTP Request Plugin Guide](https://docs.rapid7.com/insightconnect/http-requests)
```

**Review by TKuzas-R7 — APPROVED** [review]
Beautiful 👍🏻

**Pull request merged**